prediction_market_agent_tooling/data_download/langfuse_data_downloader.py

```
@@ -0,0 +1,280 @@
+            market_id
+        )
+
+        if not market or not market.resolution:
+            raise ValueError(f"No resolution found for market: {market_id}")
+
+        return market.resolution
+
+    except Exception as e:
+        raise ValueError(
+            f"Failed to fetch {market_type} market {market_id} resolution: {e}"
+        )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)